app/api/routes/chat.py

```diff
@@ -56,8 +56,23 @@ def query(chat_request: ChatRequest):
 
 @router.post("/learn", status_code=204)
 def learn(learn_request: LearnRequest):
-    learn_collection.add(
-        documents=[learn_request.text],
-        ids=[uuid.uuid4().hex],
-    )
+
+    SPLIT_INTO = 4  # Chunk size of quarter a paragraph is the best based on our preliminary experiment
+
+    documents_paragraphs = learn_request.text.split("\n")
```

Comment:
`nit` can you add a comment saying that the input text should contain only one paragraph for each line? You can add above this line, or as a docstring https://peps.python.org/pep-0257/#one-line-docstrings

Reply:
since we don't have validations on input right now, and the structure is not obvious for everyone

Reply:
I see!  That is right, I will add that too.
Would it also be better if I added a comment that the input length must be at least 4?

Reply:
yes, that would be helpful to mention the limitation, so that others might be able to refactor later

Reply:
ok! I will add that too then :)